File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    char *text = get_string("Text: ");
    int letter = 0, sentence = 0, word = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letter++;
        }
        if (isspace(text[i]))
        {
            word++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentence++;
        }
    }
    double L, S;
    int index;
    L = (double) letter / word * 100;
    S = (double) sentence / word * 100;
    index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
    return 0;
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")
    letter, sentence, word = 0, 0, 1
    for c in text:
        if c.isalpha():
            letter += 1

        if c.isspace():
            word += 1

        if c == '.' or c == '!' or c == '?':
            sentence += 1

    L = letter / word * 100
    S = sentence / word * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index >= 1 and index <= 16:
        print(f"Grade {index}\n")

    elif index < 1:
        print("Before Grade 1\n")

    else:
        print("Grade 16+\n")

    return 0

if __name__ == "__main__":
    main()